sharding/collator/smc.go

```diff
@@ -0,0 +1,30 @@
+package collator
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/sharding"
+	"github.com/ethereum/go-ethereum/sharding/client"
+)
+
+// joinCollatorPool checks if the account is a collator in the SMC. If
+// the account is not in the set, it will deposit 100ETH into contract.
+func joinCollatorPool(c client.Client) error {
```

Comment:
This is a Collator specific function. Should we move it under `collator.go` and delete `smc.go` completely?

Reply:
Sure, sounds good to me